index.js

```diff
@@ -4,140 +4,219 @@ window.addEventListener("load", () => {
   const eqButton = document.querySelector(".eButton");
   const numDisplay = document.querySelector(".numDisplay");
   const opArray = ["add", "subtract", "multiply", "divide"];
-  let currentString = numDisplay.textContent;
+  let currentString = "";
+  //This should always have STRING data. numDisplay, despite the name, should also always have STRING data.
   let firstNum = "";
+  //This should always have NUMBER data, OR a BLANK STRING (bad practice - figure out a better solution later).
   let currentOp = "";
+  //This is STRING.
+  let clearState = "ac";
+  //This is STRING.
   let errLock = false;
+  //This is BOOLEAN.
+
+  function clearCheck() {
+    //There's gotta be a more elegant way of doing this, right?
```

Comment:
Are you referring to this function being called many times in many different places? If so, I'd say the more elegant way would be having a render function that updates all the visual parts at the same time, which in this case would just be the display and the clear button. That render function would be called after every action. By the way, that's pretty much the core tenet of React.

Reply:
Noted - excited to get to React implementation.